integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/utils/PublisherUtils.java

```diff
@@ -209,6 +212,49 @@ public static String createAPI(APIRequest apiRequest, RestAPIPublisherImpl publi
         }
     }
 
+    /**
+     * creates a SOAP API with given WSDL definition
+     *
+     * @param apiRequest          An APIRequest object
+     * @param apimArtifactsIndex  APIM artifacts categorization index
+     * @param apiFileContent      API definition file content
+     * @param publisherRestClient Instance of RestAPIPublisherImpl
+     * @return ID of the created API
+     * @throws ApiException If importWSDLSchemaDefinition fails
+     * @throws IOException  If WSDL definition file read fails
+     */
+    public static String createSoapApiFromWsdl(APIRequest apiRequest, String apimArtifactsIndex,
+                                               String apiFileContent, RestAPIPublisherImpl publisherRestClient)
+            throws ApiException, IOException {
+        String apiId = null;
+        Path apisLocation = Paths.get(Utils.getTargetDirPath() + TestConstant.TEST_RESOURCES_PATH +
+                APIM_ARTIFACTS_FOLDER + apimArtifactsIndex + APIS_FOLDER);
+        if (apiRequest.getType().equals(TestConstant.API_TYPES.SOAP)) {
+            Path wsdlFilePath = Paths.get(apisLocation + "/" + new JSONObject(apiFileContent).get("wsdlUrl"));
+            File file = new File(wsdlFilePath.toString());
+            WSDLValidationResponseDTO wsdlValidationResponseDTO =
+                    publisherRestClient.validateWsdlDefinition(null, file);
+            if (wsdlValidationResponseDTO.isIsValid()) {
+                apiRequest.setWsdl(Files.readString(wsdlFilePath));
+                JSONObject additionalPropertiesObj = new JSONObject();
```

Comment:
Since these aren't additional, shall we rename to apiJson or something similar?

Reply:
fixed in https://github.com/wso2/product-microgateway/pull/2972/commits/5c91048f9fa451e4b2f2d91dac704d3f6b755cd3